ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/Endpoint.kt

```diff
@@ -178,16 +184,23 @@ internal class Endpoint(
                     }
                 }
 
-                if (!secure) return@connect connection
+                val connection = socket.connection()
+                if (!secure) {
+                    return@connect connection
+                }
 
                 try {
-                    return connection.tls(coroutineContext) {
+                    if (proxy?.type == ProxyType.HTTP && requestData.url.protocol == URLProtocol.HTTPS) {
```

Comment:
What about secure WebSocket `wss://`? Is it already erased to HTTPS at this point? Shouldn't we use the `secure` flag here? Also, isn't it always the case at this point, as we are doing an early return a few lines above?

Reply:
updated